### PR TITLE
docs(build-cli): Update type test documentation for 0.45

### PR DESCRIPTION
**Pull request by tylerbutler**
There's been some confusion about how typetest generation and management works, so I've updated the type test docs in the repo to reflect the behavior and intended design as of 0.45.0.

I am very open to changing the design in the future, but I want to make sure the current behavior is documented so everyone can be on the same page when we discuss possible improvements.

### CODE REVIEW

File: build-tools/packages/build-cli/docs/typetestDetails.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
 # Further information about `generate typetests`
 
-The `generate typetests` command generates type compatibility tests based on the individual package settings in
-package.json.
+_This documentation is accurate as of build-tools 0.45.0._
+
+The `typetests` `generate:typetests` commands are used to generate and manage type compatibility tests based on the
```
Comment (tylerbutler) [inline build-tools/packages/build-cli/docs/typetestDetails.md:5]:
```suggestion
The `typetests` and `generate:typetests` commands are used to generate and manage type compatibility tests based on the
```

File: build-tools/packages/build-cli/docs/typetestDetails.md [hunk 2]
```diff
@@ -20,154 +22,73 @@ Type test generation is primarily configured in the package.json file of the pac
...
-version that matches the range. This effectively pins the version to a specific version while allowing it to be updated
-manually as needed by running type test preparation again.
-
-This is functionally similar to what a lockfile does, but this provides us with an extra level of control so we don't
-rely on lockfiles for type test package versions, because we have found bugs in lockfile handling related to npm
-aliases, which the type test infrastructure relies on.
+> [!NOTE]
+> The `generate:typetests` command is designed to produce a single output file - the type tests - per invocation, and to
+> generate tests for a single entrypoint. If you want to generate tests for both the alpha entrypoint and the legacy
+> entrypoint, for example, you will need to use `generate:typetests` twice, once for each entrypoint.
```
Comment (tylerbutler) [inline build-tools/packages/build-cli/docs/typetestDetails.md:94]: I think this is the piece that has maybe confused people the most. I think we could make the setting an array and accept multiple values in the CLI flag in a future release, if we think that's more efficient or better than this design.

To note why I implemented it this way originally - one benefit to separate files is that it makes it easier to have different policies that apply to some type tests but not others. This is mostly relevant for CODEOWNERS, where we might want all changes to legacy type tests to be reviewed by a specific group.

**Review by ChumpChief — APPROVED** [review]
Makes sense to me!

**Comment by tylerbutler** [comment]
Thanks for the review, @ChumpChief!

**Pull request merged**